lib/functoria/misc.mli

```diff
@@ -35,3 +35,25 @@ end
 module Name : sig
   val ocamlify : string -> string
 end
+
+module Eq : sig
+  type (_, _) eq = Eq : ('a, 'a) eq | NotEq : ('a, 'b) eq
+
+  val andeq : ('a, 'b) eq -> ('c, 'd) eq -> ('a * 'c, 'b * 'd) eq
```

Comment:
do we need to expose `andeq` and `andb` ?

Reply:
They are used in the implementation of `Impl.equal`

Reply:
aaand, they are gone.